deploy/canary/Dockerfile

```diff
@@ -1,10 +0,0 @@
-FROM golang:latest
```

Comment:
The `Dockerfile` has moved in this PR to the top level dir & you can use just `make docker-build` to build the image.